airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/AbstractBigQueryDestinationAcceptanceTest.java

```diff
@@ -181,6 +184,21 @@ protected void setUpBigQuery() throws IOException {
     dataset = BigQueryDestinationTestUtils.initDataSet(config, bigquery, datasetId);
   }
 
+  protected void removeOldNamespaces() {
+    // todo (cgardens) - hardcoding to testing project to de-risk this running somewhere unexpected.
+    bigquery.listDatasets("dataline-integration-testing", DatasetListOption.all())
```

Comment:
just curious if we've tried this, I think we have thousands of datasets which won't ever get cleared up by this since they don't follow the naming convention, so iterating through all datasets every time seems like it might be costly

Reply:
we can do a one-time manual cleanup.  I'll volunteer for that once this gets merged

Reply:
I think the bigquery way to do this would be instead to just add two labels to our datasets during creation, one to add to the [DatasetListOption.labelFilter](https://cloud.google.com/java/docs/reference/google-cloud-bigquery/latest/com.google.cloud.bigquery.BigQuery.DatasetListOption#com_google_cloud_bigquery_BigQuery_DatasetListOption_labelFilter_java_lang_String_) method for querying like `purpose:acceptanceTest` since it needs to be a KV pair and the second which includes the time and can be fetched with `dataset.getLabels()` that is like `testedOn: "YYYYmmDD"` (with the actual date values) and then optionally delete

This I think alleviates my concern about iterating through all datasets every time

Reply:
random follow up for later - we may want to make sure all the datasets we care about are label, and delete any that are not

Reply:
@jbfbell i deleted 60K of them a few days ago. will blow up any remaining ones once this hits. so we should be able to get out of the world where there only namespaces in the BQ instance are test ones.

I've intentionally picked a solution that isn't BQ-specific. Ideally we can reuse the same convention in other destinations.

Reply:
ok sounds good, definitely understand not going with a bq specific implementation.